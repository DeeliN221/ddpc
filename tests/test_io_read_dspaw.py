"""Tests for the read functions in ddpc.io.read.dspaw_as."""

from pathlib import Path

from ddpc.io.read import dspaw_as
from loguru import logger


def test_read(snapshot):
    """Tests reading various as files."""
    for s in sorted(Path(f"{__file__}/../structures").resolve().iterdir()):
        if s.name.endswith(".as"):
            atoms = dspaw_as.read(s)
            logger.info(atoms)
            assert f"{s.name}\n{atoms}" == snapshot
