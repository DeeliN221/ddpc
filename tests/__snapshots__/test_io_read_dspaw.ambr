# serializer version: 1
# name: test_read
  '''
  all.as
  Atoms(symbols='S4Mo2', pbc=True, cell=[[3.19031572, 0.0, 0.0], [-1.59515786, 2.76289446, 0.0], [0.0, 0.0, 14.87900448]], initial_magmoms=...)
  '''
# ---
# name: test_read.1
  '''
  fixxyz.as
  Atoms(symbols='S4Mo2', pbc=True, cell=[[3.19031572, 0.0, 0.0], [-1.59515786, 2.76289446, 0.0], [0.0, 0.0, 14.87900448]])
  '''
# ---
# name: test_read.2
  '''
  fixxyzmagxyz.as
  Atoms(symbols='S4Mo2', pbc=True, cell=[[3.19031572, 0.0, 0.0], [-1.59515786, 2.76289446, 0.0], [0.0, 0.0, 14.87900448]], initial_magmoms=...)
  '''
# ---
# name: test_read.3
  '''
  lat.as
  Atoms(symbols='S4Mo2', pbc=True, cell=[[3.19031572, 0.0, 0.0], [-1.59515786, 2.76289446, 0.0], [0.0, 0.0, 14.87900448]])
  '''
# ---
# name: test_read.4
  '''
  mag.as
  Atoms(symbols='Ni2O2', pbc=True, cell=[[4.1684, 2.0842, 2.0842], [2.0842, 4.1684, 2.0842], [2.0842, 2.0842, 4.1684]], initial_magmoms=...)
  '''
# ---
# name: test_read.5
  '''
  magxyz.as
  Atoms(symbols='Ni2O2', pbc=True, cell=[[4.1684, 2.0842, 2.0842], [2.0842, 4.1684, 2.0842], [2.0842, 2.0842, 4.1684]], initial_magmoms=...)
  '''
# ---
